omegaconf/_utils.py

```diff
@@ -556,18 +556,18 @@ def none_as_any(t: Optional[Type[Any]]) -> Union[Type[Any], Any]:
         is_optional = obj._is_optional()
         kt = none_as_any(obj._metadata.key_type)
         vt = none_as_any(obj._metadata.element_type)
-        if ref_type is Any and kt is Any and vt is Any and obj._value() != "???":
-            pass
+        if ref_type is Any and kt is Any and vt is Any and not obj._is_missing():
+            ref_type = Any  # type: ignore
```

Comment:
what about  Container._is_none() ?
it's another case where you should not use ref_type Any (and add a test to cover it).